src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { AppMenuActionId, AppMenuActionType, AppMenuRegistry } from '@kbn/discover-utils';
+import { DATA_QUALITY_LOCATOR_ID, DataQualityLocatorParams } from '@kbn/deeplinks-observability';
+import { OBSERVABILITY_THRESHOLD_RULE_TYPE_ID } from '@kbn/rule-data-utils';
+import { isOfQueryType } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import { AppMenuExtensionParams } from '../../../..';
+import type { RootProfileProvider } from '../../../../profiles';
+import { ProfileProviderServices } from '../../../profile_provider_services';
+
+export const createGetAppMenu =
+  (services: ProfileProviderServices): RootProfileProvider['profile']['getAppMenu'] =>
+  (prev) =>
+  (params) => {
+    const prevValue = prev(params);
+
+    return {
+      appMenuRegistry: (registry) => {
+        // Register custom link actions
+        registerDatasetQualityLink(registry, services);
+        // Register alerts sub menu actions
+        registerCreateSLOAction(registry, services, params);
+        registerCustomThresholdRuleAction(registry, services, params);
+
+        return prevValue.appMenuRegistry(registry);
+      },
+    };
+  };
+
+const registerDatasetQualityLink = (
+  registry: AppMenuRegistry,
+  { share, timefilter }: ProfileProviderServices
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: i18n.translate('discover.observabilitySolution.appMenu.datasets', {
+          defaultMessage: 'Data sets',
+        }),
+        testId: 'discoverAppMenuDatasetQualityLink',
+        onClick: ({ onFinishAction }) => {
+          const refresh = timefilter.getRefreshInterval();
+          const { from, to } = timefilter.getTime();
+
+          dataQualityLocator.navigate({
+            filters: {
+              timeRange: {
+                from: from ?? 'now-24h',
+                to: to ?? 'now',
+                refresh,
+              },
+            },
+          });
+
+          onFinishAction();
+        },
+      },
+    });
+  }
+};
+
+const registerCustomThresholdRuleAction = (
```

Comment:
It looks a warning is shown when opening this in ES|QL mode:
![CleanShot 2024-11-08 at 00 06 00@2x](https://github.com/user-attachments/assets/c4761f57-6656-491e-893d-3513bca2479f)

I'm guessing the rule type doesn't support ES|QL, and in that case do you think we should still show it?

Reply:
Your guess is right, the ES|QL query is not accepted by this rule, IMO the warning gracefully handles this case but I believe is still worth it for a user to have the option to create the rule starting from the current data source.